infrastructure/templates/public-api/ci/azure-pipelines.yml

```diff
@@ -1,12 +1,21 @@
 trigger: none
 
 parameters:
-  - name: deployContainerApp
-    displayName: Can we deploy the Container App yet? This is dependent on the PostgreSQL Flexible Server being set up and having users manually added.
-    default: true
-  - name: updatePsqlFlexibleServer
+  - name: deploySharedPrivateDnsZones
+    displayName: Do the shared Private DNS Zones need creating or updating?
+    default: false
+  - name: deployPsqlFlexibleServer
     displayName: Does the PostgreSQL Flexible Server require any updates? False by default to avoid unnecessarily lengthy deploys.
     default: false
```

Comment:
Hey yeah, I have investigated in the past, especially around PSQL which is a real time drain.  Didn't come away with anything concrete - apart from the tags, an identical subsequent deploy will post all of the same properties but result in a lot of churn.

Seems to be that according to the docs, it's not ARM's responsibility to ensure that subsequent updates to already-deployed resources don't result in a bunch of rework.  ARM just orchestrates stuff and then sends all of the deployment parameters off to the relevant resource provider (e.g. PostgreSQL Flexible Server).  It's up to the individual resource providers to then determine what to do with those.  An example is here: https://github.com/Azure/bicep/discussions/7001.

Reply:
Ah thought it'd be something like that. Bit of a shame, but I guess we're stuck with it. 

Does put a bit of a case forward for using Terraform, which maintains state and would be able to avoid making unnecessary changes. Guess we're not changing to that anytime soon though!

Reply:
Haha yes it would be nice for sure - I guess that the resource providers'll only improve over time... right...?